contracts/src/token/link_token.cairo

```diff
@@ -14,21 +14,39 @@ mod LinkToken {
 
     use zeroable::Zeroable;
 
-    use starknet::ContractAddress;
-    use starknet::class_hash::ClassHash;
-
-    use openzeppelin::token::erc20::ERC20;
     use openzeppelin::token::erc20::interface::{IERC20, IERC20Dispatcher, IERC20DispatcherTrait};
     use chainlink::libraries::token::erc677::ERC677;
     use chainlink::libraries::ownable::{Ownable, IOwnable};
     use chainlink::libraries::upgradeable::{Upgradeable, IUpgradeable};
 
+    use openzeppelin::token::erc20::ERC20Component;
+    use starknet::ContractAddress;
+    use starknet::class_hash::ClassHash;
+
+    component!(path: ERC20Component, storage: erc20, event: ERC20Event);
+
+    #[abi(embed_v0)]
+    impl ERC20Impl = ERC20Component::ERC20Impl<ContractState>;
+    #[abi(embed_v0)]
+    impl ERC20MetadataImpl = ERC20Component::ERC20MetadataImpl<ContractState>;
+    impl ERC20InternalImpl = ERC20Component::InternalImpl<ContractState>;
+
     const NAME: felt252 = 'ChainLink Token';
     const SYMBOL: felt252 = 'LINK';
 
     #[storage]
     struct Storage {
         _minter: ContractAddress,
+
+        #[substorage(v0)]
+        erc20: ERC20Component::Storage
+    }
+
+    #[event]
+    #[derive(Drop, starknet::Event)]
+    enum Event {
+        #[flat]
```

Comment:
what does this flat macro do?

Reply:
it'll flatten the subvariants into the parent event rather than having it nested